rmgpy/data/kinetics/family.py

```diff
@@ -1464,33 +1483,65 @@ def generateReactions(self, reactants, products=None, prod_resonance=True):
             specified reactants and products within this family.
             Degenerate reactions are returned as separate reactions.
         """
-        reactionList = []
-        
-        # Forward direction (the direction in which kinetics is defined)
-        reactionList.extend(self.__generateReactions(reactants, products=products, forward=True, prod_resonance=prod_resonance))
-        
-        if not self.ownReverse and self.reversible:
-            # Reverse direction (the direction in which kinetics is not defined)
-            reactionList.extend(self.__generateReactions(reactants, products=products, forward=False, prod_resonance=prod_resonance))
+        reaction_list = []
+
+        # Check that reactants is a list of :class:Molecule objects or a list of lists of :class:Molecule objects,
+        # and verify that only reactive molecules react.
+        invalid_types = False
+        reactive_reactants = []
+        for reactant in reactants:
+            if isinstance(reactant, Molecule):
+                if reactant.reactive:
+                    reactive_reactants.append(reactant)
+                else:
+                    logging.debug("Not generating reactions for the unreactive molecule {0}".format(reactant))
+            elif isinstance(reactant, list):
+                react_list = []
+                for react in reactant:
+                    if isinstance(react, Molecule):
+                        if react.reactive:
+                            react_list.append(react)
+                        else:
+                            logging.debug("Not generating reactions for the unreactive molecule {0}".format(react))
+                    else:
+                        invalid_types = True
+                reactive_reactants.append(react_list)
+            else:
+                invalid_types = True
+            if invalid_types:
+                raise ValueError("'reactants' must be a list of :class:Molecule or a list of lists containing"
+                                 " :class:Molecule objects")
 
-        return reactionList
 
-    def addReverseAttribute(self, rxn):
+        reactants = [reactant if isinstance(reactant, list) else [reactant] for reactant in reactants]
+        for reactant in reactants:
+            if not all([mol.reactive for mol in reactant]):
```

Comment:
I might not have been clear about the list of lists. In the typical case, this takes reactants as `[reactantA, reactantB]`. However, it can also properly handle `[[A_res1, A_res2], [B_res1, B_res2]]`, where the sublists are lists of resonance structures for each reactant. I think I was also confused in my original comment about this part; it was correct to check that all the reactants are reactive.

Now that I think about it more though, it may be unnecessary do the check here at all. Since you already check in `__generateReactions()` where the actual generation is done, I don't think you need to change this method at all.

Reply:
I see. I'll revert the original change